### PR TITLE
[FEQ]Yaswanth/FEQ-1233/Migration from circle-ci to github work actions

### PR DESCRIPTION
**Pull request by yaswanth-deriv**
As a developer, I must migrate from CircleCI to GitHub Workflows, ensuring its smooth execution without failures and successful deployment

### CODE REVIEW

File: .github/actions/send_slack_notifications/action.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: send_slack_notifications
+description: Send Slack notifications
+inputs:
+  SLACK_WEBHOOK_URL:
+    description: 'Slack webhook URL'
+    required: true
+  status:
+    description: 'Job status'
+    required: true
+  release_type:
```
Comment (ali-hosseini-deriv) [inline .github/actions/send_slack_notifications/action.yml:10]:
```suggestion
  RELEASE_TYPE:
```

File: .github/workflows/release-production.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        - name: Build
+          run: npm run build
+        - name: Test
+          run: npm run test
+        - name: Publish to Cloudflare Staging
+          uses: ./.github/actions/publish_to_cloudflare_production
+
+    send_slack_notification:
+        name: Send Slack notification
+        runs-on: Runner_16cores_Deriv-app
```
Comment (ali-hosseini-deriv) [inline .github/workflows/release-production.yml:26]:
```suggestion
        runs-on: ubuntu-latest
```

File: .github/actions/invalidate_master_cache/action.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+name: invalidate_npm_cache
+description: Invalidate the Master NPM cache
+runs:
+  using: composite
+  steps:
+  - name: save_cache
+    uses: actions/cache/save@v3
+    with:
+      path: ./node_modules
+      key: v1-deps-{{ checksum "package-lock.json" }}
```
Comment (ali-hosseini-deriv) [inline .github/actions/invalidate_master_cache/action.yml:10]:
```suggestion
      key: v1-deps-{{ checksum "Gemfile" }}
```

File: .github/actions/npm_install_from_cache/action.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+name: npm_install_from_cache
+description: Install npm packages from cache
+runs:
+  using: composite
+  steps:
+    - name: Cache dependencies
+      id: cache-node-modules
+      uses: actions/cache/restore@v3
+      with:
+        key: v1-deps-{{ checksum "package-lock.json" }}
```
Comment (ali-hosseini-deriv) [inline .github/actions/npm_install_from_cache/action.yml:10]:
```suggestion
        key: v1-deps-{{ checksum "Gemfile" }}
```

File: .github/actions/npm_install_from_cache/action.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        path: node_modules
+    - name: Install npm dependencies
+      if: steps.cache-node-modules.outputs.cache-hit != 'true'
+      run: npm install
```
Comment (ali-hosseini-deriv) [inline .github/actions/npm_install_from_cache/action.yml:14]: its not a node app 
please check the circleci file and follow the steps from there.
```suggestion
      run: bundle install --path ./modules
```

**Review by ali-hosseini-deriv — CHANGES_REQUESTED** [review]

File: .github/actions/npm_install_from_cache/action.yml (hunk 5 above)
Comment (yaswanth-deriv) [reply .github/actions/npm_install_from_cache/action.yml:14]: @ali-hosseini-deriv Sorry ..I should keep this in draft..I still haven't finished it yet.

File: .github/actions/versioning/action.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    required: false
+outputs:
+  version:
+    description: Version
+runs:
+  using: composite
+  steps:
+    - run:
+        name: Tag build
+        run: echo "<< parameters.target_branch >> $(date -u +'%Y-%m-%dT%H:%M:%SZ')" > _site/version
```
Comment (ali-hosseini-deriv) [inline .github/actions/versioning/action.yml:15]:
```suggestion
        run: echo "${GITHUB_REF#refs/heads/} $(date -u +'%Y-%m-%dT%H:%M:%SZ')" > _site/version
```

**Review by ali-hosseini-deriv — APPROVED** [review]

**Pull request merged**